boards/arm/qemu_cortex_m3/qemu_cortex_m3_defconfig

```diff
@@ -1,9 +1,33 @@
 CONFIG_ARM=y
-CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=12000000
-CONFIG_SOC_TI_LM3S6965=y
+CONFIG_SOC_FAMILY_ARM=y
+CONFIG_SOC_SERIES_MPS2=y
+CONFIG_SOC_MPS2_AN385=y
 CONFIG_BOARD_QEMU_CORTEX_M3=y
+CONFIG_CORTEX_M_SYSTICK=y
+CONFIG_RUNTIME_NMI=y
+
+# GPIOs
+CONFIG_GPIO=y
+
+# PinMuxing
+CONFIG_PINMUX=y
+
+# Serial
 CONFIG_CONSOLE=y
 CONFIG_UART_CONSOLE=y
 CONFIG_SERIAL=y
-CONFIG_CORTEX_M_SYSTICK=y
-CONFIG_UART_STELLARIS=y
+CONFIG_UART_CMSDK_APB=y
+
+# Watchdog
+CONFIG_WATCHDOG=y
+
+CONFIG_I2C=y
+
+#userspace
+CONFIG_ARCH_HAS_USERSPACE=y
+CONFIG_ARM_MPU=y
+CONFIG_MAIN_STACK_SIZE=2048
+CONFIG_IDLE_STACK_SIZE=1024
```

Comment:
So if the minimum MPU region is sized 1K, and the guard area is taken from the existing stack size instead of added to it, does this mean that the effective stack size for the idle stack is zero? How does this work?

Reply:
yep what you said is true. This is quite strange. I never hit a fault when I ran this. I was supposed to get a stacking error from the MPU. 
On a different note: There are a few issues on the MPU stack sentinel front. Persistent even when I up the stack size to 4k.